tidy3d/components/data/monitor_data.py

```diff
@@ -575,6 +578,251 @@ class FluxTimeData(MonitorData):
     flux: FluxTimeDataArray
 
 
+# pylint: disable=too-many-public-methods
+class DiffractionData(MonitorData):
+    """Data associated with a :class:`.DiffractionMonitor`:
+    complex components of diffracted far fields.
+
+    Example
+    -------
+    >>> f = np.linspace(1e14, 2e14, 10)
+    >>> orders_x = list(range(-4, 5))
+    >>> orders_y = list(range(-6, 7))
+    >>> pol = ["s", "p"]
+    >>> coords = dict(f=f, orders_x=orders_x, orders_y=orders_y, polarization=pol)
+    >>> values = (1+1j) * np.random.random((len(orders_x), len(orders_y), len(pol), len(f)))
+    >>> field = DiffractionDataArray(values, coords=coords)
+    >>> monitor = DiffractionMonitor(
+    ...     center=(1,2,3), size=(np.inf,np.inf,0), freqs=f, name='diffraction',
+    ...     orders_x=orders_x, orders_y=orders_y
+    ...     )
+    >>> data = DiffractionData(
+    ...     monitor=monitor, L=field, N=field, sim_size=[1,1], bloch_vecs=[1,2]
+    ... )
+    """
+
+    monitor: DiffractionMonitor
+
+    sim_size: Tuple[float, float] = pd.Field(
+        ...,
+        title="Simulation size",
+        description="Simulation sizes in the local x and y directions.",
+    )
+
+    bloch_vecs: Tuple[float, float] = pd.Field(
+        ...,
+        title="Bloch vectors",
+        description="Bloch vectors along the local x and y directions in bandstructure units.",
```

Comment:
Sorry, so is "bandstructure units" basically in units of the unit cell length in x and y? in other words, is `bloch_vecs[i]` in units of `sim_size[i]`? perhaps we should clarify a bit more.